rero_ils/modules/documents/serializers.py

```diff
@@ -52,51 +52,64 @@ def post_process_serialize_search(self, results, pid_fetcher):
         """Post process the search results."""
         # Item filters.
         viewcode = request.args.get('view')
-        if viewcode != current_app.config.get(
-                'RERO_ILS_SEARCH_GLOBAL_VIEW_CODE'
-        ):
-            view_id = Organisation.get_record_by_viewcode(viewcode)['pid']
+        if viewcode is not None:
+            # Add availibility
             records = results.get('hits', {}).get('hits', {})
             for record in records:
                 metadata = record.get('metadata', {})
                 available = Document.get_record_by_pid(
                     metadata.get('pid')).is_available(viewcode)
                 metadata['available'] = available
-                items = metadata.get('items', [])
-                if items:
-                    output = []
-                    for item in items:
-                        if item.get('organisation')\
-                                .get('organisation_pid') == view_id:
-                            output.append(item)
-                    record['metadata']['items'] = output
-        else:
-            records = results.get('hits', {}).get('hits', {})
-            for record in records:
-                metadata = record.get('metadata', {})
-                available = Document.get_record_by_pid(
-                    metadata.get('pid')).is_available(viewcode)
-                metadata['available'] = available
-
+            if viewcode != current_app.config.get(
+                    'RERO_ILS_SEARCH_GLOBAL_VIEW_CODE'
+            ):
+                # Compute availibility using viewcode
+                # and filter items by viewcode
+                viewcode_organisation_pid = \
+                    Organisation.get_record_by_viewcode(viewcode)['pid']
+                for record in records:
```

Comment:
Maybe using of https://www.programiz.com/python-programming/methods/built-in/filter makes this code clearer.